### ISSUE TITLE
system.reboot does not return on EL7

### ISSUE THREAD

**Issue by lloesche** [saltstack/salt#24297]
When running a system.reboot usually the minions will return a success state. On CentOS7 however they don't (get to) respond

```
[root@salt1.vm.ca1 pillar]# salt 'srv*.hw.onpremtest.ca1' system.reboot
srv4.hw.onpremtest.ca1:
    Minion did not return. [No response]
srv1.hw.onpremtest.ca1:
    Minion did not return. [No response]
srv3.hw.onpremtest.ca1:
    Minion did not return. [No response]
srv5.hw.onpremtest.ca1:
    Minion did not return. [No response]
srv2.hw.onpremtest.ca1:
    Minion did not return. [No response]
```

**Comment by jfindlay** [comment]
@lloesche, thanks for the report.

**Comment by jfindlay** [comment]
@lloesche, what is the output of `salt --versions-report` and `salt 'srv*.hw.onpremtest.ca1' test.versions_report` if the minion versions are different?  Thanks.

**Comment by lloesche** [comment]
I believe they're all running the latest EPEL packages.

```
[root@salt1.vm.ca1 ~]# salt --versions-report
           Salt: 2015.5.0
         Python: 2.7.5 (default, Jun 17 2014, 18:11:42)
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
 msgpack-python: 0.4.6
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 14.3.1
           RAET: Not Installed
            ZMQ: 3.2.5
           Mako: Not Installed
[root@salt1.vm.ca1 ~]# salt 'srv*.hw.onpremtest.ca1' test.versions_report
srv5.hw.onpremtest.ca1:
               Salt: 2015.5.0
             Python: 2.7.5 (default, Jun 17 2014, 18:11:42)
             Jinja2: 2.7.2
           M2Crypto: 0.21.1
     msgpack-python: 0.4.6
       msgpack-pure: Not Installed
           pycrypto: 2.6.1
            libnacl: Not Installed
             PyYAML: 3.10
              ioflo: Not Installed
              PyZMQ: 14.3.1
               RAET: Not Installed
                ZMQ: 3.2.5
               Mako: Not Installed
srv3.hw.onpremtest.ca1:
               Salt: 2015.5.0
             Python: 2.7.5 (default, Jun 17 2014, 18:11:42)
             Jinja2: 2.7.2
           M2Crypto: 0.21.1
     msgpack-python: 0.4.6
       msgpack-pure: Not Installed
           pycrypto: 2.6.1
            libnacl: Not Installed
             PyYAML: 3.10
              ioflo: Not Installed
              PyZMQ: 14.3.1
               RAET: Not Installed
                ZMQ: 3.2.5
               Mako: Not Installed
srv4.hw.onpremtest.ca1:
               Salt: 2015.5.0
             Python: 2.7.5 (default, Jun 17 2014, 18:11:42)
             Jinja2: 2.7.2
           M2Crypto: 0.21.1
     msgpack-python: 0.4.6
       msgpack-pure: Not Installed
           pycrypto: 2.6.1
            libnacl: Not Installed
             PyYAML: 3.10
              ioflo: Not Installed
              PyZMQ: 14.3.1
               RAET: Not Installed
                ZMQ: 3.2.5
               Mako: Not Installed
srv1.hw.onpremtest.ca1:
               Salt: 2015.5.0
             Python: 2.7.5 (default, Jun 17 2014, 18:11:42)
             Jinja2: 2.7.2
           M2Crypto: 0.21.1
     msgpack-python: 0.4.6
       msgpack-pure: Not Installed
           pycrypto: 2.6.1
            libnacl: Not Installed
             PyYAML: 3.10
              ioflo: Not Installed
              PyZMQ: 14.3.1
               RAET: Not Installed
                ZMQ: 3.2.5
               Mako: Not Installed
srv2.hw.onpremtest.ca1:
               Salt: 2015.5.0
             Python: 2.7.5 (default, Jun 17 2014, 18:11:42)
             Jinja2: 2.7.2
           M2Crypto: 0.21.1
     msgpack-python: 0.4.6
       msgpack-pure: Not Installed
           pycrypto: 2.6.1
            libnacl: Not Installed
             PyYAML: 3.10
              ioflo: Not Installed
              PyZMQ: 14.3.1
               RAET: Not Installed
                ZMQ: 3.2.5
               Mako: Not Installed
```

**Comment by basepi** [comment]
This is semi-expected behavior. Restarting the machine interrupts the minion process and so it never gets a chance to return. Not sure of the fix here.

**Comment by lloesche** [comment]
@basepi well I get that. I just figured I report it because the behaviour is not consistent across systems. On Ubuntu the call does return before the system is rebooted.

**Comment by basepi** [comment]
Ah, thanks for the clarification, I didn't read closely enough.

**Comment by mbochenk** [comment]
Have the same behaviour on EL7. 
Looking into documentation there is at least one example, suggesting that expected behaviour would be return a value: [salt.states.saltmod.html#salt.states.saltmod.wait_for_event](https://docs.saltstack.com/en/latest/ref/states/all/salt.states.saltmod.html#salt.states.saltmod.wait_for_event)
We have to apply some changes that require full reboot of the system and it would be helpful if system.reboot could return something at first.
Documentation also does not state that this behaviour is semi-expected: [salt.modules.system.html#salt.modules.system.reboot](https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.system.html#salt.modules.system.reboot)

**Comment by ealphonse** [comment]
Hi,

#31557 is related.

It used to return successfully on Ubuntu with `2015.8.10` but it is no longer the case with `2016.3.4`.

As stated by @mbochenk, the doc indicates it should return a value:
https://docs.saltstack.com/en/latest/ref/states/all/salt.states.saltmod.html#salt.states.saltmod.wait_for_event

For now, I use: `salt minion cmd.run "/usr/bin/nohup /bin/bash -c 'set -x && sleep 5 && systemctl reboot' >>/var/log/salt/minion 2>&1 &"`

**Comment by ealphonse** [comment]
For the record, the use case comes from: #6792

**Comment by anitakrueger** [comment]
I know this was closed as stale, but it is still happening in 2019.2.0.
Is there a workaround to catch the reboot in an orchestration even if the `system.reboot` comes back as failed?
